src/payment-flows/nonce.js

```
@@ -0,0 +1,107 @@
+/* @flow */
+
+import { FUNDING } from '@paypal/sdk-constants/src/funding';
+
+import { payWithNonce } from '../api';
+import { getLogger, promiseNoop } from '../lib';
+
+import type { PaymentFlow, PaymentFlowInstance } from './types';
+
+function setupNonce() {
+// pass
+}
+
+function isNonceEligible({ props, serviceData }) : boolean {
+    const { paymentMethodNonce } = props;
+    // eslint-disable-next-line no-console
+    console.log('nonce eligibility check', paymentMethodNonce);
+
+    const { wallet } = serviceData;
+
+    // eslint-disable-next-line no-console
+    console.log('wallet', wallet);
+
+    if (!wallet) {
+        return false;
+    }
+
+    // Ensure wallet instruments are branded and have a valid tokenID.
+    if (wallet.card.instruments.length === 0 ||
```

We should also check if there is anything in wallet where `tokenID === paymentMethodNonce`, else return false

---

good call

---

Thinking about it more- I am curious if this will be the same as `!wallet.card.instruments.find(({ tokenID })  => (tokenID === paymentMethodID)`. We do this check in `isNoncePaymentEligible`, but would be good to fail early

---

Yeah good to short-circuit early 👍